geoSegment/geoSegment.js

```diff
@@ -167,26 +169,56 @@ async function emptyOrMakeOutputDirectories(segments) {
  * @param {ScheduledSessionData} scheduledSessionData
  * @param {SessionSeriesData} sessionSeries
  */
-async function mergeScheduledSessionAndSessionSeriesAndWrite(scheduledSessionData, sessionSeries) {
+async function mergeScheduledSessionAndSessionSeriesAndWrite(scheduledSessionData, sessionSeries, segments) {
   // Link ScheduledSession with SessionSeries
   const mergedScheduledSessionData = {
     ...sessionSeries.superEvent,
     ...dissocPath(['superEvent'], sessionSeries),
     ...scheduledSessionData,
-    ...{ name: (sessionSeries.superEvent && sessionSeries.superEvent.name) || sessionSeries.name || scheduledSessionData.name },
-    ...{
-      'oo:localStartDate': moment(scheduledSessionData.startDate).tz('Europe/London').format('DD/MM/YYYY'),
-      'oo:localStartTime': moment(scheduledSessionData.startDate).tz('Europe/London').format('HH:mm'),
-      'oo:localEndDate': moment(scheduledSessionData.endDate).tz('Europe/London').format('DD/MM/YYYY'),
-      'oo:localEndTime': moment(scheduledSessionData.endDate).tz('Europe/London').format('HH:mm'),
-    },
+    name: (sessionSeries.superEvent && sessionSeries.superEvent.name) || sessionSeries.name || scheduledSessionData.name,
+    'oo:localStartDate': moment(scheduledSessionData.startDate).tz('Europe/London').format('DD/MM/YYYY'),
+    'oo:localStartTime': moment(scheduledSessionData.startDate).tz('Europe/London').format('HH:mm'),
+    'oo:localEndDate': moment(scheduledSessionData.endDate).tz('Europe/London').format('DD/MM/YYYY'),
+    'oo:localEndTime': moment(scheduledSessionData.endDate).tz('Europe/London').format('HH:mm'),
   };
 
+  const scheduledSessionGeoSegments = (() => {
+    const generateSegments = (physicalLocationGeo) => {
+      const temp = [];
+      for (const segment of segments) {
+        const segmentRadiusInMeters = segment.radius * 1000;
+        const distanceBetweenGeos = geolib.getDistance(
+          { latitude: physicalLocationGeo.latitude, longitude: physicalLocationGeo.longitude },
+          { latitude: segment.latitude, longitude: segment.longitude },
+        );
+        if (distanceBetweenGeos <= segmentRadiusInMeters) {
+          temp.push(segment.identifier);
+        }
+      }
+      return temp;
+    };
+
+    if (isOfflineEvent(mergedScheduledSessionData)) {
+      if (mergedScheduledSessionData.location && mergedScheduledSessionData.location.geo) {
+        return generateSegments(mergedScheduledSessionData.location.geo);
+      }
+      return []; // no location data
+    } if (mergedScheduledSessionData.eventAttendanceMode === 'https://schema.org/OnlineEventAttendanceMode') {
+      if (mergedScheduledSessionData['beta:affiliatedLocation'] && mergedScheduledSessionData['beta:affiliatedLocation'].geo) { // affiliate location present
+        return generateSegments(mergedScheduledSessionData['beta:affiliatedLocation'].geo);
+      }
+      return segments.map((_) => _.identifier);
```

Comment:
Could you leave a lil comment here explaining why this happens

Reply:
which part ?